src/gennylib/include/gennylib/PoolFactory.hpp

```diff
@@ -45,7 +45,7 @@ class PoolFactory {
     };
 
 public:
-    PoolFactory(std::string_view uri);
+    PoolFactory(std::string_view uri, mongocxx::options::client& clientOpts);
```

Comment:
Since `PoolFactory` owns the `mongocxx::options::client` options you should just take it by value and `std::move()` it into the constructor for places it is being used. I'd recommend having it default to `mongocxx::options::client{}` and discarding your changes to PoolFactory_test.cpp.